pybop/builders/pybamm.py

```diff
@@ -53,7 +53,7 @@ def set_simulation(
         self._model = model.new_copy()
         self._initial_state = initial_state
         self._build_on_eval = build_on_eval
```

Comment:
do we need another method on the builder to set the number of threads to use? i.e. so users have the option of turning off multithreading or only using a subset of their cpus